scipy/stats/tests/test_morestats.py

```diff
@@ -2460,106 +2463,81 @@ class TestCircFuncs:
-                      [357, 9, 8, 358, 4, 356]])
-
-        V1 = stats.circvar(x, high=360)
-        V2 = stats.circvar(x.ravel(), high=360)
-        assert_allclose(V1, V2, rtol=1e-11)
-
-        V1 = stats.circvar(x, high=360, axis=1)
-        V2 = [stats.circvar(x[i], high=360) for i in range(x.shape[0])]
-        assert_allclose(V1, V2, rtol=1e-11)
-
-        V1 = stats.circvar(x, high=360, axis=0)
-        V2 = [stats.circvar(x[:, i], high=360) for i in range(x.shape[1])]
-        assert_allclose(V1, V2, rtol=1e-11)
-
-    def test_circstd_axis(self):
-        x = np.array([[355, 5, 2, 359, 10, 350],
-                      [351, 7, 4, 352, 9, 349],
-                      [357, 9, 8, 358, 4, 356]])
-
-        S1 = stats.circstd(x, high=360)
-        S2 = stats.circstd(x.ravel(), high=360)
-        assert_allclose(S1, S2, rtol=1e-11)
-
-        S1 = stats.circstd(x, high=360, axis=1)
-        S2 = [stats.circstd(x[i], high=360) for i in range(x.shape[0])]
-        assert_allclose(S1, S2, rtol=1e-11)
-
-        S1 = stats.circstd(x, high=360, axis=0)
-        S2 = [stats.circstd(x[:, i], high=360) for i in range(x.shape[1])]
-        assert_allclose(S1, S2, rtol=1e-11)
+        x = np.asarray([0.12675364631578953] * 10 + [0.12675365920187928] * 100)
+        circstat = test_func(xp.asarray(x))
+        normal = xp.asarray(numpy_func(x))
+        xp_assert_close(circstat, normal, atol=2e-8)
+
+    @pytest.mark.parametrize('circfunc', [stats.circmean,
+                                          stats.circvar,
+                                          stats.circstd])
+    def test_circmean_axis(self, xp, circfunc):
+        x = xp.asarray([[355, 5, 2, 359, 10, 350],
+                        [351, 7, 4, 352, 9, 349],
+                        [357, 9, 8, 358, 4, 356.]])
+        res = circfunc(x, high=360)
+        ref = circfunc(xp.reshape(x, (-1,)), high=360)
+        xp_assert_close(res, xp.asarray(ref), rtol=1e-14)
+
+        res = circfunc(x, high=360, axis=1)
+        ref = [circfunc(x[i, :], high=360) for i in range(x.shape[0])]
+        xp_assert_close(res, xp.asarray(ref), rtol=1e-14)
+
+        res = circfunc(x, high=360, axis=0)
+        ref = [circfunc(x[:, i], high=360) for i in range(x.shape[1])]
+        xp_assert_close(res, xp.asarray(ref), rtol=1e-12)
 
     @pytest.mark.parametrize("test_func,expected",
                              [(stats.circmean, 0.167690146),
                               (stats.circvar, 0.006455174270186603),
                               (stats.circstd, 6.520702116)])
-    def test_circfuncs_array_like(self, test_func, expected):
-        x = [355, 5, 2, 359, 10, 350]
-        assert_allclose(test_func(x, high=360), expected, rtol=1e-7)
+    def test_circfuncs_array_like(self, test_func, expected, xp):
+        x = xp.asarray([355, 5, 2, 359, 10, 350.])
+        rtol = 2e-5 if is_torch(xp) else 1e-7
+        xp_assert_close(test_func(x, high=360), xp.asarray(expected), rtol=rtol)
```

Comment:
<details>
<summary> torch GPU local failure summary </summary>

```
scipy/stats/tests/test_morestats.py:2527: in test_circfuncs_array_like
    xp_assert_close(test_func(x, high=360), xp.asarray(expected), rtol=rtol)
E   AssertionError: Scalars are not close!
E   
E   Expected 0.16769014298915863 but got 0.16770131886005402.
E   Absolute difference: 1.1175870895385742e-05 (up to 0 allowed)
E   Relative difference: 6.664596198780912e-05 (up to 2e-05 allowed)
```

</details>

Increasing `rtol` to `6.7e-5` was the tightest I could manage locally to have it pass with `torch` on GPU.

Reply:
Ok, I think with the changed default tolerance for `float32` (`eps**0.5`) this will pass.